File: cash.c
#include <cs50.h>
#include <math.h>
#include <stdio.h>

int main(void)
{
    float n;

    do
    {
        n = get_float("Change owed: ");

    }
    while (n < 0);

    // int tt = qt + dm + nk + ct;
    int ct = round(n * 100); // debugged
    int qt = 0;
    int dm = 0;
    int nk = 0;    

    // calculate quarter coins first
    qt = round(ct / 25);
    // use modulo to replace "ct = ct - qt * 25"
    ct = ct % 25;

    if (ct == 0)
    {
        printf("%d\n", qt + dm + nk + ct);
    }
    else
    {
        // calculate dimes
        dm = round(ct / 10);
        ct = ct % 10;

        if (ct == 0)
        {
            printf("%d\n", qt + dm + nk + ct);
        }
        else
        {
            // calculate nickels and pennies
            nk = round(ct / 5);
            ct = ct % 5;

            printf("%d\n", qt + dm + nk + ct);
        }
    }
}

File: week6/sentimental/cash.py
from cs50 import get_float 

c = 0

while True:
    try:
        c = float(get_float("Change owed: "))
    except ValueError:
        continue
    if c < 0:
        continue
    else:
        break
# cents
c = int(c * 100)
# dimes
q = int(c / 25)
c = c % 25
if c == 0:
    print(q)
else:
    # dimes
    d = int(c / 10)
    c = c % 10
    if c == 0:
        print(q + d)
    else:
        # nickels
        n = int(c / 5)
        c = c % 5
        print(q + d + n + c)
        